presto-native-execution/presto_cpp/main/PrestoMain.cpp

```diff
@@ -31,7 +34,14 @@ int main(int argc, char* argv[]) {
   PRESTO_SHUTDOWN_LOG(INFO) << "Exiting main()";
 }
 
-// Initialize singleton for the reporter.
-folly::Singleton<facebook::velox::BaseStatsReporter> reporter([]() {
-  return new facebook::velox::DummyStatsReporter();
-});
+// Initialize singleton for the reporter
+folly::Singleton<facebook::velox::BaseStatsReporter> reporter(
+    []() -> facebook::velox::BaseStatsReporter* {
+#ifdef PRESTO_ENABLE_PROMETHEUS_REPORTER
+      return facebook::presto::prometheus::PrometheusStatsReporter::
+          createPrometheusReporter()
```

Comment:
This is causing an implementation header (prometheus-reporter) to be included which is not ideal.
Let's add a function inside `RuntimeMetricsReporter.h/cpp` that returns the right implementation and invoke that here.

Reply:
I wonder if we need to initialize the `facebook::velox::DummyStatsReporter();`. If not, we can define the reporter inside prometheus-reporter.

Reply:
@karteekmurthys I actually agree with @majetideepak. Thinking about more, we don't need to even initialize DummyReporter in the main function. We can initialize PrometheusStatsReporter or DummyReporter inside Prometheus folder (kind of what you did before).

Reply:
@amitkdutta Are you sure about this? I thought you faced issues on Meta side when the reporter was declared outside PrestoMain.

Reply:
@karteekmurthys Yes,  thats because inside Meta there is a different main function which also create a statsreporter singletone. There are 2 ways to skip it:
- declare in the main function (like here). Passing a different main function then avoid the issue of duplicate registration
- declare inside prometheus folder. 
In both cases, we need to be able to build without including prometheus, meaning we should not have the header included anywhere where the macro `PRESTO_ENABLE_PROMETHEUS_REPORTER` isn't enabled.

I think we can skip registering the dummy reporter, and only register Prometheus reporter when PRESTO_ENABLE_PROMETHEUS_REPORTER is enabled.